src/Validator/RefundUnitsCommandValidator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Validator;
+
+use Sylius\RefundPlugin\Checker\OrderRefundingAvailabilityCheckerInterface;
+use Sylius\RefundPlugin\Command\RefundUnits;
+use Sylius\RefundPlugin\Exception\CreditMemoCommentTooLongException;
+use Sylius\RefundPlugin\Exception\OrderNotAvailableForRefundingException;
+
+final class RefundUnitsCommandValidator implements RefundUnitsCommandValidatorInterface
+{
+    private const MAX_COMMENT_LENGTH = 256;
+
+    /** @var OrderRefundingAvailabilityCheckerInterface */
+    private $orderRefundingAvailabilityChecker;
+
+    public function __construct(OrderRefundingAvailabilityCheckerInterface $orderRefundingAvailabilityChecker)
+    {
+        $this->orderRefundingAvailabilityChecker = $orderRefundingAvailabilityChecker;
+    }
+
+    /**
+     * @throws OrderNotAvailableForRefundingException|CreditMemoCommentTooLongException
+     */
+    public function validate(RefundUnits $command): void
+    {
+        if (!$this->orderRefundingAvailabilityChecker->__invoke($command->orderNumber())) {
```

Comment:
Probably could be written as `($this->orderRefundingAvailabilityChecker)($command->orderNumber())`